lib/net/ssh/proxy/command.rb

```diff
@@ -93,7 +93,7 @@ def io.recv(size)
             result = read_nonblock(size)
           rescue IO::WaitReadable, Errno::EINTR
             timeout_in_seconds = 20
-            if IO.select([self], nil, [self], timeout_in_seconds) == nil
+            if wait_readable(timeout_in_seconds) == nil
```

Comment:
https://github.com/ruby/ruby/blob/trunk/ext/io/wait/wait.c#L118-L128 (trunk)
https://github.com/ruby/ruby/blob/v2_0_0_648/ext/io/wait/wait.c#L95-L104 (v2.0.0-p648)

Although 2.0 returns false if read is not possible (check macro), both return nil on timeout. Hence, for 2.0 compatibility, the nil check, as it correctly translates to a timeout check.